src/components/MenuBar/MenuBarToggleButton.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+    IconButton,
+    IconButtonProps,
+    Tooltip,
+    TooltipTypeMap,
+} from "@mui/joy";
+
+import Check from "@mui/icons-material/Check";
+import Close from "@mui/icons-material/Close";
+
+
+interface MenuBarToggleButtonProps extends IconButtonProps {
+    tooltipPlacement?: TooltipTypeMap["props"]["placement"];
+    isActive?: boolean;
+    onIcon?: React.ReactNode;
+    offIcon?: React.ReactNode;
+}
+
+/**
+ * A toggle button for use in the menu bar.
+ *
+ * @param props
+ * @param props.title Tooltip title.
+ * @param props.tooltipPlacement Tooltip position.
+ * @param props.onIcon Icon when active.
+ * @param props.offIcon Icon when inactive.
+ * @param props.rest Other IconButton props.
+ * @param props.isActive
+ * @return JSX.Element
+ */
+const MenuBarToggleButton = ({
+    tooltipPlacement,
+    title,
+    isActive = false,
+    onIcon = <Check/>,
+    offIcon = <Close/>,
+    ...rest
+}: MenuBarToggleButtonProps) => {
+    return (
+        <Tooltip
+            placement={tooltipPlacement ?? "bottom"}
```

Comment:
would it be fine to directly supply `tooltipPlacement` if it's `undefined`?
```suggestion
            placement={tooltipPlacement}
```

Reply:
I tried to remove `?? "bottom"`, but I have an error: `TS2769: No overload matches this call.`

Reply:
Right, now i remember - we have `exactOptionalPropertyTypes
` turned on, but the monaco lib doesn't handle this well

Reply:
i proposed a solution at https://github.com/y-scope/yscope-log-viewer/pull/223/files#r2034595711

Reply:
Fixed!